### ISSUE TITLE
Cannot resolve github domain when using modules

### ISSUE THREAD

**Issue by aacecandev** [weaveworks/tf-controller#264]
Hi,

I've found a possible bug when using the tf-runner with Terraform modules. When I try to deploy a lambda function using a Github hosted Terraform module, the tf-runner pod initializes, but all of sudden throws an error.

```
{"level":"info","ts":"2022-07-10T16:33:48.277Z","logger":"controller.terraform","msg":"runner is running","reconciler group":"infra.contrib.fluxcd.io","reconciler kind":"Terraform","name":"aws-lambda","namespace":"flux-system"}
{"level":"info","ts":"2022-07-10T16:33:48.277Z","logger":"controller.terraform","msg":"getting source","reconciler group":"infra.contrib.fluxcd.io","reconciler kind":"Terraform","name":"aws-lambda","namespace":"flux-system"}
{"level":"info","ts":"2022-07-10T16:33:48.277Z","logger":"controller.terraform","msg":"setting up terraform","reconciler group":"infra.contrib.fluxcd.io","reconciler kind":"Terraform","name":"aws-lambda","namespace":"flux-system"}
{"level":"info","ts":"2022-07-10T16:33:48.318Z","logger":"controller.terraform","msg":"write backend config: ok","reconciler group":"infra.contrib.fluxcd.io","reconciler kind":"Terraform","name":"aws-lambda","namespace":"flux-system"}
{"level":"info","ts":"2022-07-10T16:33:48.320Z","logger":"controller.terraform","msg":"new terraform","reconciler group":"infra.contrib.fluxcd.io","reconciler kind":"Terraform","name":"aws-lambda","namespace":"flux-system","workingDir":"/tmp/flux-system-aws-lambda/iac/terraform"}
{"level":"error","ts":"2022-07-10T16:33:48.697Z","logger":"controller.terraform","msg":"error in terraform setup","reconciler group":"infra.contrib.fluxcd.io","reconciler kind":"Terraform","name":"aws-lambda","namespace":"flux-system","error":"error running Init: rpc error: code = Unknown desc = exit status 1\n\nError: Failed to download module\n\nCould not download module \"lambda_function\" (lambda.tf:1) source code from\n\"git::https://github.com/terraform-aws-modules/terraform-aws-lambda?ref=v3.3.1\":\nerror downloading\n'https://github.com/terraform-aws-modules/terraform-aws-lambda?ref=v3.3.1':\n/usr/bin/git exited with 128: Cloning into\n'.terraform/modules/lambda_function'...\nfatal: unable to access\n'https://github.com/terraform-aws-modules/terraform-aws-lambda/': Could not\nresolve host: github.com\n.\n\n\nError: Failed to download module\n\nCould not download module \"lambda_function\" (lambda.tf:1) source code from\n\"git::https://github.com/terraform-aws-modules/terraform-aws-lambda?ref=v3.3.1\":\nerror downloading\n'https://github.com/terraform-aws-modules/terraform-aws-lambda?ref=v3.3.1':\n/usr/bin/git exited with 128: Cloning into\n'.terraform/modules/lambda_function'...\nfatal: unable to access\n'https://github.com/terraform-aws-modules/terraform-aws-lambda/': Could not\nresolve host: github.com\n.\n\n"}
```

I'm not sure if I'm missing something when configuring the Chart. I've tried two different container images:

- ghcr.io/weaveworks/tf-runner:v0.9.5-rc.3
- ghcr.io/weaveworks/tf-runner:v0.10.0-rc.6

For debugging purposes, I've spin up a busybox pod and made a nslookup to github.com successfully. Also I've tried the same procedure using the above images into a pod shell, and it resolves OK

```
/ $ nslookup github.com
Server:         10.96.0.10
Address:        10.96.0.10:53

Name:   github.com
Address: 140.82.121.3
```

I've also tried the same running a tf-runner container in my machine, and it also resolves the domain. It only fails when executing through the default entrypoint.

**Comment by chanwit** [comment]
Hi @aacecandev 
Thank you for reporting. 

I cannot reproduce this problem. Are you still facing it?

**Comment by aacecandev** [comment]
@chanwit I'm still facing it. I didn't attach the source code, but it is public.

https://github.com/aacecandev/flux-testing
https://github.com/aacecandev/tf-lambda

Latest commit commented out all the module code and I tried a simple S3 bucket and it worked.

**Comment by chanwit** [comment]
@aacecandev could you test it against v0.10.1 please?
It might be caused by HOME being the read-only filesystem, which was fixed to be writeable in v0.10.1.

**Comment by LappleApple** [comment]
Closing as we're now several versions ahead that have addressed this issue.
